### ISSUE TITLE
Content Files Not Copying When emitEntryPoint == false

### ISSUE THREAD

**Issue by user4794170** [dotnet/cli#1410]
Hello. 

I'm building a library (.dll). The dll needs a few .exe's that are in the project directory that I need copied to the output folder on build. I've found that if emitEntryPoint is set to false (as appropriate for a dll), the content is not copied to the output folder. 

I've created a repro batch file that demonstrates the issue. In the demo, I've swapped out my three .exes for a single .txt, but the problem remains the same. My project is dnx451, but the issue repros for both dnx451 and dnxcore50. 

**How can I setup my project.json for my library so that it copies content, such as data files, to the output folder?**

Thanks!

---
#### repro.bat

```
cd c:\
mkdir Repro
cd Repro
dotnet new
echo "My Important Data" > "myImportantFile.txt"

echo "Working, with emitEntryPoint == true, update project.json to project.json-working"
start /wait notepad project.json
dotnet restore
dotnet build
echo "Note text file is included"
dir c:\Repro\bin\debug\dnxcore50
pause
del /q /s c:\Repro\bin\

echo "Broken, with emitEntryPoint == false, update project.json to project.json-broken"
start /wait notepad project.json
dotnet restore
dotnet build
echo "Note text file is not included"
dir c:\Repro\bin\debug\dnxcore50
pause

cd c:\
del /q /s c:\Repro
```
#### project.json-working (with emitEntryPoint == true)

```
{
    "version": "1.0.0-*",
    "compilationOptions": {
        "emitEntryPoint": true
    },
    "dependencies": { "NETStandard.Library": "1.0.0-rc2-23727" },
    "frameworks": {
        "dnxcore50": {}
    },
    "contentFiles": ["myImportantFile.txt"]
}
```
#### project.json-broken (with emitEntryPoint == false)

```
{
    "version": "1.0.0-*",
    "compilationOptions": {
        "emitEntryPoint": false
    },
    "dependencies": { "NETStandard.Library": "1.0.0-rc2-23727" },
    "frameworks": {
        "dnxcore50": {}
    },
    "contentFiles": ["myImportantFile.txt"]
}
```

**Comment by davidfowl** [comment]
The bigger question is how you plan to consume this. There's 2 scenarios:

- what happens when you make a nuget package 
- what happens when you use as as a project reference 

What does the code look like that's reading those files? What's the assumption about where they are located

**Comment by user4794170** [comment]
Hi @davidfowl. 
Thanks for taking the time to reply.

I'm working on a shared library that uses Selenium (and therefore needs the per browser driver executables in the output folder). I won't be making Nuget packages but this library will be referenced by other projects, including Test library projects as well as Console applications. The Test and Console applications will also need the exe's in the output folder so that Selenium will run correctly. 

I've been doing this with .NET 4.5.1 using the cproj system so far, where I've told VS to copy the executables to the output folder on build but I'm currently in the process of migrating the code to dotnet-core, which is where I need the same functionality. 

I'm not sure I entirely understand the question about code location - I'm planning on adding the shared library as a project reference where appropriate. 

Let me know if I can provide any other context to help.
Thanks.

**Comment by user4794170** [comment]
Just an update: with dnx451, build doesn't seem to copy content files, regardless of whether emitEntryPoint is set to true or false - this is especially a problem for me since Selenium won't be coming to dotnet-core for a long while.

**Comment by user4794170** [comment]
Hi @davidfowl. Can you please comment on this further? This is a blocking issue for me - if we can't do this, then I need to switch back to .NET 4.6 w/ csproj. I'm hoping to know as soon as possible as I'm the one pushing our organization to move to .NET Core and there are several people blocked waiting on this. Thanks!

**Comment by user4794170** [comment]
Looks like there is a partial hacky workaround. In my Library project, I've added an empty Program.cs with a Main and enabled emitEntryPoint to true. This allows the data exes to get copied to the output folder. Unfortunately, doing the same for the Test project doesn't copy over the exes from the output of Library project (the way it works in csproj) so my tests still fail anyway. On top of this, the exes only get copied out if I build the Library project directly. Building Test project with the Library project referenced, builds the Library project, but doesn't include the exe files.

**Comment by livarcocc** [comment]
@pakrym Where are we on honoring Content for dependencies?

**Comment by pakrym** [comment]
In progress on Project to Package, but not for P2P

**Comment by andschwa** [comment]
Wouldn't Project to Project be more important? We have a test project that depends on the main project, and since the content of the main project (as a dependency of the test project) doesn't get honored, it totally breaks the tests (we work-around by copying the files in the wrapper script...).

**Comment by piotrpMSFT** [comment]
@davidfowl mentioned to me that a P2P must be indistinguishable from a package dependency. If this is not true, folks will need to rethink how they test in order to validate what they ship. Are we saying that a P2P is no longer steadfastly equivalent to a package dependency?

**Comment by pakrym** [comment]
It would be the same for p2p, just not implemented yet.

**Comment by andschwa** [comment]
I think this issue is a duplicate of #753.

**Comment by piotrpMSFT** [comment]
@andschwa, right!
